src/main/features/claimant-response/routes/payment-plan.ts

```diff
@@ -52,14 +52,24 @@ class PaymentPlanPage extends AbstractPaymentPlanPage<DraftClaimantResponse> {
   buildPostSubmissionUri (req: express.Request, res: express.Response): string {
     const claim: Claim = res.locals.claim
     const draft: Draft<DraftClaimantResponse> = res.locals.draft
-    const courtDecision = CourtDecisionHelper.createCourtDecision(claim, draft)
+    const courtDecision: DecisionType = CourtDecisionHelper.createCourtDecision(claim, draft)
+    const claimResponse: FullAdmissionResponse | PartialAdmissionResponse = claim.response as FullAdmissionResponse | PartialAdmissionResponse
 
     const externalId: string = req.params.externalId
     switch (courtDecision) {
-      case DecisionType.COURT:
-      case DecisionType.DEFENDANT: {
+      case DecisionType.COURT: {
         return Paths.courtOfferPage.evaluateUri({ externalId: externalId })
       }
+      case DecisionType.DEFENDANT: {
+        if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+          return Paths.courtOfferPage.evaluateUri({ externalId: externalId })
+        }
+
+        if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+          return Paths.courtOfferedSetDatePage.evaluateUri({ externalId: externalId })
+        }
```

Comment:
The above logic seems duplicate in `src/main/features/claimant-response/routes/payment-date.ts` Prefer to refactor this out

Reply:
`  buildPostSubmissionUri (req: express.Request, res: express.Response): string {` is duplicated in `routes/payment-date.ts` and `routes/payment-plan.ts`

Reply:
I will do this when i am working on immediately. Not yet sure what happens there, its my next story.